Sources/Starknet/Accounts/StarknetAccount.swift

```diff
@@ -86,100 +86,100 @@ public class StarknetAccount: StarknetAccountProtocol {
         var nonce: Felt
         var maxFee: Felt
 
         if let paramsNonce = params.nonce {
             nonce = paramsNonce
         } else {
-            nonce = try await getNonce()
+            nonce = try await getNonce().send()
         }
 
         if let paramsMaxFee = params.maxFee {
             maxFee = paramsMaxFee
         } else {
-            let feeEstimate = try await estimateFeeV1(calls: calls, nonce: nonce)
+            let feeEstimate = try await estimateFeeV1(calls: calls, nonce: nonce).send()[0]
             maxFee = feeEstimate.toMaxFee()
         }
 
         let params = StarknetInvokeParamsV1(nonce: nonce, maxFee: maxFee)
         let signedTransaction = try signV1(calls: calls, params: params, forFeeEstimation: false)
 
-        return try await provider.addInvokeTransaction(signedTransaction)
+        return provider.addInvokeTransaction(signedTransaction)
     }
 
-    public func executeV3(calls: [StarknetCall], params: StarknetOptionalInvokeParamsV3) async throws -> StarknetInvokeTransactionResponse {
+    public func executeV3(calls: [StarknetCall], params: StarknetOptionalInvokeParamsV3) async throws -> StarknetRequest<StarknetInvokeTransactionResponse, AddInvokeTransactionParams> {
         var nonce: Felt
         var resourceBounds: StarknetResourceBoundsMapping
 
         if let paramsNonce = params.nonce {
             nonce = paramsNonce
         } else {
-            nonce = try await getNonce()
+            nonce = try await getNonce().send()
         }
 
         if let paramsResourceBounds = params.resourceBounds {
             resourceBounds = paramsResourceBounds
         } else {
-            let feeEstimate = try await estimateFeeV3(calls: calls, nonce: nonce)
+            let feeEstimate = try await estimateFeeV3(calls: calls, nonce: nonce).send()[0]
             resourceBounds = feeEstimate.toResourceBounds()
         }
 
         let params = StarknetInvokeParamsV3(nonce: nonce, l1ResourceBounds: resourceBounds.l1Gas)
         let signedTransaction = try signV3(calls: calls, params: params, forFeeEstimation: false)
 
-        return try await provider.addInvokeTransaction(signedTransaction)
+        return provider.addInvokeTransaction(signedTransaction)
     }
 
-    public func executeV1(calls: [StarknetCall], estimateFeeMultiplier: Double) async throws -> StarknetInvokeTransactionResponse {
-        let nonce = try await getNonce()
-        let feeEstimate = try await estimateFeeV1(calls: calls, nonce: nonce)
+    public func executeV1(calls: [StarknetCall], estimateFeeMultiplier: Double) async throws -> StarknetRequest<StarknetInvokeTransactionResponse, AddInvokeTransactionParams> {
+        let nonce = try await getNonce().send()
+        let feeEstimate = try await estimateFeeV1(calls: calls, nonce: nonce).send()[0]
```

Comment:
This seems problematic to me. Someone can try to batch two `executeV1` requests like this, and they will both have the same nonce. 

I'm thinking we should think of the ways to prevent users from shooting themselves in the foot by either:
- Disallowing batching for some methods, specifically those that rely on sending other requests to generate a request
- Adding this info to docstring/README

WDYT?

Reply:
I think disallowing it in code would be a safer choice. Maybe let's address it in another PR, as it is not blocking core functionality.